### PR TITLE
fix(KFLUXBUGS-1197): component relationship remove button

### PR DESCRIPTION
**Pull request by caugello**
## Fixes 
Missing remove button on component relationship modal

## Description
Fixes [KFLUXBUGS-1197](https://issues.redhat.com//browse/KFLUXBUGS-1197)

## Type of change

- [x] Bugfix

## Browser conformance: 

- [x ] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

**Review by sahil143 — CHANGES_REQUESTED** [review]
@caugello Thank you for the PR.

Alignment and color of  remove icon seems off,
image

should be like this
image


When a relationship is removed, a API request should also be sent to update the relationship for the removed component. In the above screenshot If I remove the second relationship, A request should be sent to update the component `go-devfile-sample-vuln`. I guess we should compare the OldData with the new data from form here https://github.com/caugello/hac-dev/blob/bd7d19099be9c39c220baa5289d6142ce1025d86/src/components/ComponentRelation/utils.ts#L73-L74

File: src/components/ComponentRelation/__tests__/utils.spec.ts [hunk 1]
```diff
@@ -1,5 +1,71 @@
 import { ComponentRelationNudgeType } from '../type';
-import { componentRelationValidationSchema, transformNudgeData } from '../utils';
+import {
+  componentRelationValidationSchema,
+  computeNudgeDataChanges,
+  transformNudgeData,
+} from '../utils';
+
+describe('transformNudgeData', () => {
```
Comment (sahil143) [inline src/components/ComponentRelation/__tests__/utils.spec.ts:8]: unit tests for `transformNudgeData` already exist, check below Line 70. Can we remove these? or Are these different?

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/ComponentRelation/__tests__/utils.spec.ts (hunk 1 above)
Comment (caugello) [reply src/components/ComponentRelation/__tests__/utils.spec.ts]: It's a dupe, removed it.

**Review by sahil143 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**